kubernetes-client-api/src/test/java/io/fabric8/kubernetes/client/utils/KubernetesResourceUtilTest.java

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+ package io.fabric8.kubernetes.client.utils;
+
+import static io.fabric8.kubernetes.client.utils.KubernetesResourceUtil.inferListType;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.api.model.PodList;
+import org.junit.jupiter.api.Test;
+
+
+public class KubernetesResourceUtilTest {
+  /**
+   * To verify null in Thread.currentThread().getContextClassLoader() would be handled in loadRelated()
+   */
+  @Test
+  void testNullContextClassLoader() {
+    Thread.currentThread().setContextClassLoader(null);
```

Comment:
Please do this in a try finally that restores the context classloader.

Reply:
Good point. Fixed.